File: docs/examples/js/src/data_management_delete_records.mjs
import { Client } from "reduct-js";
import assert from "node:assert";

// Create a client instance, then get or create a bucket
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });
const bucket = await client.getOrCreateBucket("bucket");

// Send some records to the "entry-1" entry
const timestamp = BigInt(Date.now()) * 1000n;
let record = await bucket.beginWrite("entry-1",  timestamp);
await record.write("Some binary data");

record = await bucket.beginWrite("entry-1",  timestamp + 1000_000n);
await record.write("Some more binary data");

// Delete a sole record
await bucket.removeRecord("entry-1", timestamp);

// Delete a batch of records
const batch = await bucket.beginRemoveBatch("entry-1");
batch.addOnlyTimestamp(timestamp);  // already deleted, so this error will be in the errors map
batch.addOnlyTimestamp(timestamp + 1000_000n);
const errors = await batch.write();

assert(errors.size === 1);
console.log(errors);

File: versioned_docs/version-1.20.x/examples/go/src/data_management_delete_records.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Send some records to the "go-example"
	ts := time.Now().UnixMicro()
	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts + 1,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	// Delete a single record
	err = bucket.RemoveRecord(context.Background(), "go-example", ts)
	if err != nil {
		panic(err)
	}

	// Delete a batch of records
	batch := bucket.BeginRemoveBatch(context.Background(), "go-example")
	// Add the timestamp of the already deleted record
	batch.AddOnlyTimestamp(ts) // already deleted, so this error will be in the errors map
	batch.AddOnlyTimestamp(ts + 1)

	errors, err := batch.Write(context.Background())
	if err != nil {
		panic(err)
	}

	if len(errors) != 1 {
		panic("Expected 1 error, but got " + string(len(errors)))
	}
	for ts, err := range errors {
		println("Error deleting record with timestamp", ts, ":", err.Error())
	}
}
